components/net/http_loader.rs

```diff
@@ -242,56 +239,67 @@ fn is_schemelessy_same_site(site_a: &ImmutableOrigin, site_b: &ImmutableOrigin)
 }
 
 /// <https://w3c.github.io/webappsec-referrer-policy/#strip-url>
-fn strip_url(mut referrer_url: ServoUrl, origin_only: bool) -> Option<ServoUrl> {
+fn strip_url_for_use_as_referrer(mut url: ServoUrl, origin_only: bool) -> Option<ServoUrl> {
     const MAX_REFERRER_URL_LENGTH: usize = 4096;
-    if referrer_url.scheme() == "https" || referrer_url.scheme() == "http" {
-        {
-            let referrer = referrer_url.as_mut_url();
-            referrer.set_username("").unwrap();
-            referrer.set_password(None).unwrap();
-            referrer.set_fragment(None);
-            // Limit `referer` header's value to 4k <https://github.com/w3c/webappsec-referrer-policy/pull/122>
-            if origin_only || referrer.as_str().len() > MAX_REFERRER_URL_LENGTH {
-                referrer.set_path("");
-                referrer.set_query(None);
-            }
+    // Step 2
+    if url.is_local_scheme() {
+        return None;
+    }
+    // Step 3-6
+    {
+        let url = url.as_mut_url();
+        url.set_username("").unwrap();
+        url.set_password(None).unwrap();
+        url.set_fragment(None);
+        if origin_only || url.as_str().len() > MAX_REFERRER_URL_LENGTH {
```

Comment:
Let's add a note that the length check is Step 6 of https://w3c.github.io/webappsec-referrer-policy/#determine-requests-referrer.